### PR TITLE
FL - Data Simulation

### PR DESCRIPTION
**Pull request by jvmancuso**
Allows users to run more TF Datasets with the FL example. Also cleans up the driving script a bit.

At one point I was trying to be able to do splitting/simulation of datasets in a generic way, but eventually conceded and limited this PR to image datasets (i.e. ones with `image` & `label` features).  This change occurred between commits `8a7b5ad` & `5242319`.

### CODE REVIEW

File: examples/federated-learning/players.py [hunk 1]
```diff
@@ -28,20 +28,20 @@ class BaseModelOwner:
...
   """
 
-  def __init__(self, player_name, local_data_file, model, loss, optimizer=None):
+  def __init__(self, player_name, local_tfrecords, model, loss, optimizer=None):
     self.player_name = player_name
 
-    self.optimizer = optimizer
     self.loss = loss
+    self.optimizer = optimizer
+    self.dataset = tf.data.TFRecordDataset(local_tfrecords)
```
Comment (mortendahl) [inline examples/federated-learning/players.py]: won't instantiation here not place the dataset on the wrong player?

File: examples/federated-learning/data_lib.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+from functools import wraps
+import logging
+import os
+from pathlib import Path
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+import tf_encrypted as tfe
+
+logger = logging.getLogger('tf_encrypted')
```
Comment (mortendahl) [inline examples/federated-learning/data_lib.py]: maybe we should add this to core TFE, say `tfe.debugging.getLogger`?

**Comment by jvmancuso** [comment]
>Okay to merge this up, but we need to define what we're trying to achieve with the FL example.

was deferring that to the TFF RFC, this work was meant to add capabilities for data scientist experimentation

File: examples/federated-learning/data_lib.py (hunk 2 above)
Comment (jvmancuso) [reply examples/federated-learning/data_lib.py]: there is some precedent for doing so from TF, as they have their own debugger module. thinking we should still keep ours separate, but we should also try to standardize how we log things against theirs (currently there are some discrepancies). either way, think that calls for a separate PR :)

Comment (jvmancuso) [reply examples/federated-learning/data_lib.py]: there is some precedent for doing so from TF, as they have their own debugger module. thinking we should still keep ours separate, but we should also try to standardize how we log things against theirs (currently there are some discrepancies). either way, think that calls for a separate PR :)

Comment (mortendahl) [reply examples/federated-learning/data_lib.py]: See https://github.com/tf-encrypted/tf-encrypted/issues/722
